near-sdk-core/src/info_extractor/attr_sig_info.rs

```diff
@@ -14,8 +14,12 @@ pub struct AttrSigInfo {
     pub non_bindgen_attrs: Vec<Attribute>,
     /// All arguments of the method.
     pub args: Vec<ArgInfo>,
-    /// Whether method can be used as initializer.
+    /// Whether the method is an initializer.
+    /// NOTE: It's non-inclusive to `is_init_once`.
     pub is_init: bool,
+    /// Whether the method is an initializer and the contract state should't exist.
+    /// NOTE: If it's true, it means `is_init` is true as well.
```

Comment:
It sounds cleaner to make it non-overlapping, e.g. `is_init` would correspond to `[init]` attribute only. Then we can have a method on this struct like `is_some_init_method`

Reply:
It's easier to manage internally because there are bunch of places that depend on `is_init`, while `is_init_once` is optional for `is_init`.